WordPress/Sniffs/WP/CapabilitiesSniff.php

```diff
@@ -0,0 +1,470 @@
+		'resume_themes'               => true,
+		'setup_network'               => true,
+		'switch_themes'               => true,
+		'unfiltered_html'             => true,
+		'unfiltered_upload'           => true,
+		'update_core'                 => true,
+		'update_https'                => true,
+		'update_languages'            => true,
+		'update_plugins'              => true,
+		'update_php'                  => true,
+		'update_themes'               => true,
+		'upgrade_network'             => true,
+		'upload_files'                => true,
+		'upload_plugins'              => true,
+		'upload_themes'               => true,
+		'view_site_health_checks'     => true,
+	);
+
+	/**
+	 * List of deprecated core capabilities.
+	 *
+	 * User Levels were deprecated in version 3.0.
+	 *
+	 * {@internal To be updated after every major release. Last updated for WordPress 6.1.0.}
+	 *
+	 * @link https://github.com/WordPress/wordpress-develop/blob/master/tests/phpunit/tests/user/capabilities.php
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array<string, string> All deprecated capabilities in core.
+	 */
+	protected $deprecated_capabilities = array(
+		'level_10' => '3.0.0',
+		'level_9'  => '3.0.0',
+		'level_8'  => '3.0.0',
+		'level_7'  => '3.0.0',
+		'level_6'  => '3.0.0',
+		'level_5'  => '3.0.0',
+		'level_4'  => '3.0.0',
+		'level_3'  => '3.0.0',
+		'level_2'  => '3.0.0',
+		'level_1'  => '3.0.0',
+		'level_0'  => '3.0.0',
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$function_name_lc = strtolower( $matched_content );
+		$function_details = $this->target_functions[ $function_name_lc ];
+
+		$parameter = PassedParameters::getParameterFromStack(
+			$parameters,
+			$function_details['position'],
+			$function_details['name']
+		);
+
+		if ( false === $parameter ) {
+			return;
+		}
+
+		// If the parameter is anything other than T_CONSTANT_ENCAPSED_STRING throw a warning and bow out.
+		$first_non_empty = null;
+		for ( $i = $parameter['start']; $i <= $parameter['end']; $i++ ) {
+			if ( isset( Tokens::$emptyTokens[ $this->tokens[ $i ]['code'] ] ) ) {
+				continue;
+			}
+
+			if ( \T_CONSTANT_ENCAPSED_STRING !== $this->tokens[ $i ]['code']
+				|| null !== $first_non_empty
+			) {
+				// Throw warning at low severity.
+				$this->phpcsFile->addWarning(
+					"Couldn't determine the value passed to the capability parameter. Please check if it matches a valid capability.",
+					$i,
+					'NeedsManualCheck',
+					array( $parameter['clean'] ),
```

Comment:
```suggestion
					array( $function_details['name'], $matched_content, $parameter['clean'] ),
```

Note: `$parameter['clean']` was unused without the `Found: ... ` addition.